### PR TITLE
Sync Acronym

### PR DESCRIPTION
**Pull request by fejan-malek**
https://github.com/exercism/php/issues/745

Below Changes done in this pull request
- Updated the Markdown file.
- New test cases added and removed tests which are not mentioned in `test.toml`

### CODE REVIEW

**Comment by fejan-malek** [comment]
@mk-mxp Should I update the example code here?

Please review thanks!

**Comment by mk-mxp** [comment]
> Please review thanks!

I hope to find time today. I'm busy with working for my living...

File: exercises/practice/acronym/AcronymTest.php [hunk 1]
```diff
@@ -31,41 +9,84 @@ public static function setUpBeforeClass(): void
...
+    /**
+     * @testdox Underscore emphasis
+     * uuid: adc12eab-ec2d-414f-b48c-66a4fc06cdef
+     */
+    public function testUnderscoreEmphasis(): void
     {
-        $phrase = 'Специализированная процессорная часть';
-        $this->assertEquals('СПЧ', acronym($phrase));
+        $this->assertEquals('TRNT', acronym('The Road _Not_ Taken'));
     }
```
Comment (mk-mxp) [inline exercises/practice/acronym/AcronymTest.php:91]: We cannot add apostrophes and underscores. The rules were not defined (and tested) before. These fail for all solutions that rely on `preg_*()` word boundary detection. And that is many.

So, please remove the last two tests (apostrophes and underscores). We only add the one with the long abbreviation and the consecutive delimiters.

Thanks!

**Review by mk-mxp — CHANGES_REQUESTED** [review]

File: exercises/practice/acronym/AcronymTest.php (hunk 1 above)
Comment (mk-mxp) [reply exercises/practice/acronym/AcronymTest.php:91]: @fejan-malek When you mark the excluded tests `include=false` in `tests.toml`, please add `comment=This invalidates existing solutions that rely on RegExp word boundary detection`.

**Comment by fejan-malek** [comment]
@mk-mxp 

`testUnderscoreEmphasis`  Is working correctly?

Should I also add in not included in `test.toml`? If yes then what is the reason because it running with the existing solution?

**Comment by mk-mxp** [comment]
No underscores, please. Reasons:

The `example.php` is an internal solution, showing that

* it is possible to solve the exercise in PHP
* a solution exists, that runs fast enough for exercisms infrastructure

It is not enough, that `example.php` can handle the new tests. We need to look at the community solutions for very common solutions.

There are currently about 350 different community solutions (see https://exercism.org/tracks/php/build > "Practice Exercises" > 106 active exercises > Acronym). These are shown partially on the exercise community solutions page (see https://exercism.org/tracks/php/exercises/acronym/solutions when you have solved the exercise / are not logged in). Many of them use `preg_*()` functions and rely on the word boundary detection (`\w` and others) of PCRE. An underscore is not seen as a word boundary by PCRE, so these solutions fail.

File: exercises/practice/acronym/AcronymTest.php [hunk 2]
```diff
@@ -31,41 +9,66 @@ public static function setUpBeforeClass(): void
...
         $this->assertEquals('HTML', acronym('HyperText Markup Language'));
     }
 
+    /**
+     * @testdox All caps word
+     * uuid: 32dd261c-0c92-469a-9c5c-b192e94a63b0
+     */
     public function testAllCapsWords(): void
     {
         $this->assertEquals('PHP', acronym('PHP: Hypertext Preprocessor'));
```
Comment (mk-mxp) [inline exercises/practice/acronym/AcronymTest.php:45]:
```suggestion
        $this->assertEquals('GIMP', acronym('GNU Image Manipulation Program'));
```

**Review by mk-mxp — CHANGES_REQUESTED** [review]
Why haven't I seen this earlier? Shame on me...

**Review by mk-mxp — APPROVED** [review]
Thank you for contribution!

**Pull request merged**